### PR TITLE
Fixes double thralls

### LINKED ISSUE #20581: Vampire thralls are listed multiple times in the end game report
image

They shouldn't be

### PR DESCRIPTION
**Pull request by ShiftyRail**
[roleissue]
Closes #20581 

I was a retard defining something when it's already being taken care off by the main proc :(
Also unatomically makes the Drop() proc of the thrall actually use the var it's defined with

### CODE REVIEW

File: code/datums/mind.dm [hunk 1]
```diff
@@ -93,12 +93,13 @@
 	if(active)
 		new_character.key = key		//now transfer the key to link the client to our new body
 
-	for (var/role in antag_roles)
-		var/datum/role/R = antag_roles[role]
-		R.PostMindTransfer(new_character, current)
-
+	var/mob/old_character = current
```
Comment (ShiftyRail) [inline code/datums/mind.dm]: I blame Git for this

File: code/datums/gamemode/role/vampire_role.dm [hunk 2]
```diff
@@ -102,13 +102,13 @@
...
-	. = ..() // Who he was, his objectives...
-	. += "Total blood collected: <b>[blood_total]</b><br/>"
-	for (var/datum/role/thrall/T in faction.members)
-		. += T.GetScoreboard()
-		. += "<br/>"
+	. = "Total blood collected: <b>[blood_total]</b><br/>"
+	. += ..() // Who he was, his objectives...
 
 /datum/role/vampire/ForgeObjectives()
+	if(!SOLO_ANTAG_OBJECTIVES)
```
Comment (ShiftyRail) [inline code/datums/gamemode/role/vampire_role.dm:109]: I'm not sure why this is showing up when it's already on the codebase

Comment (Intigracy) [reply code/datums/gamemode/role/vampire_role.dm:109]: Did you update your local bleeding-edge before making this PR?
Looks like this PR needs a rebase.

Comment (ShiftyRail) [reply code/datums/gamemode/role/vampire_role.dm:109]: This is fixed now

**Pull request merged**